rails_programming/forms_and_authentication/form_basics.md

```diff
@@ -155,27 +169,27 @@ From the Rails Guide:
 ~~~
 
 ~~~erb
-  #app/views/articles/new.html.erb
-  <%= form_for @article do |f| %>
-    <%= f.text_field :title %>
-    <%= f.text_area :body, size: "60x12" %>
+  #app/views/posts/new.html.erb
+  <%= form_with model: @post do |form| %>
+    <%= form.text_field :title %>
     <%= f.submit "Create" %>
   <% end %>
 ~~~
 
-And the generated HTML is:
+This will produce the following HTML:
 
 ~~~html
-  <form accept-charset="UTF-8" action="/articles/create" method="post">
-    <input id="article_title" name="article[title]" type="text" />
-    <textarea id="article_body" name="article[body]" cols="60" rows="12"></textarea>
-    <input name="commit" type="submit" value="Create" />
-  </form>
+  <form action="/posts" method="post" data-remote="true">
+  <input type="text" name="post[title]">
+</form>
 ~~~
 
-Note that this helper nests the Article's attributes (the hard brackets in the `name` attribute should be the dead giveaway).
+The best part about `form_with` is that if you just pass it a model object like `@post` in the example above, Rails will check for you if the object has been saved yet.  If it's a new object, it will send the form to your `#create` action.  If the object has been saved before, so we know that we're editing an existing object, it will send the object to your `#update` action instead.  This is done by automatically generating the correct URL when the form is created.  Magic!
```

Comment:
```suggestion
The best part about `form_with` is that if you just pass it a model object like `@article` in the example above, Rails will check for you if the object has been saved yet.  If it's a new object, it will send the form to your `#create` action.  If the object has been saved before, so we know that we're editing an existing object, it will send the object to your `#update` action instead.  This is done by automatically generating the correct URL when the form is created.  Magic!
```